### ISSUE TITLE
R package structure pointers

### ISSUE THREAD

**Issue by pdiakumis** [umccr/RNAsum#57]
Opening this for pointers to a potential R package skeleton for RNAsum. This should by no means be a current priority imho, since there are other more pressing issues with the full move to ICA. But worth doing going forward.

## skeleton

Most of these directories contain small and/or auto-generated files, so it's not as daunting as it appears.

- `DESCRIPTION`: this is what makes this an R pkg. Describes rnasum pkg, lists dependencies, author, license etc.
- `NAMESPACE`: all exported functions are listed here
- `R/`: holds all functions you've defined. Doesn't need to have anything, you can just keep the Rmd report in `inst/` below.
- `inst/`
  - `cli/rnasum.R`: cli script
  - `rmd/rnasum.Rmd`: Rmd report
  - `extdata/`: holds raw data
  - `img/`: holds images used in docs
  - `scripts/`: holds random scripts you want to keep
- `conda/`
  - `recipe/{meta.yaml,build.sh}`: contains conda recipe with dependencies as defined in DESCRIPTION
- `man/`: contains (auto-generated) Rd help doc files
- `tests/`: for testing your functions
- `data/`: holds rda data
- `data-raw/`: scripts to generate data in data/
- `vignettes/`: for various vignetts/articles/usecases you'd like to showcase
- `pkgdown/_pkgdown.yml`: for generating a pkgdown website
- `LICENSE`: this goes together with LICENSE.md and 'License: MIT + file LICENSE' in DESCRIPTION
- `LICENSE.md`
- `NEWS.md`: changelog
- `README.Rmd`: this gets rendered to README.md if you want R code evaluated in your README
- `README.md`
- `rnasum.Rproj`: for opening pkg in RStudio
- `.Rbuildignore`: files to ignore when building R pkg
- `.bumpversion.cfg`: for helping with version bumps
- `.pre-commit-config.yaml`: for performing small checks before committing files
- `.github/workflows/build.yml`: for building conda pkg, pkgdown doc website (which also acts as a testing suite) and docker image 

## resources

- https://usethis.r-lib.org/
- https://r-pkgs.org/
- https://pkgdown.r-lib.org/

## why?

- `devtools::install_github("umccr/RNAsum")` # covers users who don't want to use conda
  - could also be used in conjunction with something like `renv`
- `mamba install -c umccr r-rnasum` # covers users who are on HPC/want conda
- Your Docker image would need a base conda installation (as is the case currently), then you could install r-rnasum in it (which would automatically pull in its dependencies).
- You could have conda-lock files that explicitly list all the conda dependencies required for a certain conda env, which would ensure reproducibility.
- You adhere to stricter R pkg guidelines, which require all functions to explicitly list their source which would help solve difficult issues e.g. `base::sum` vs. `rapportools::sum`.
- The `pkgdown` website is a great way to showcase your tool
- For both the conda and Dockerised versions you'd have the cli script as a simple conda binary which you could call as easily as e.g. 

```
rnasum \
  --sample_name sampleA \
  --dataset TEST \
  --dragen_rnaseq path/to/test_sample_WTS_dragen_v3.9.3 \
  --report_dir path/to/RNAsum_report \
  --umccrise path/to/umccrised/sbj
```

In short: I think it makes total sense to have this as an R package going forward, but yes, it needs a week or two of work and there are a few infrastructure code chunks that need to be adjusted.

**Comment by pdiakumis** [comment]
Think we're way past this point :-)
